cli: TestTenantZip failed

cli.TestTenantZip [failed](https://tanzanite.cluster.engflow.com/invocations/default/005a74ae-29f5-436a-a47a-edf55ea12319?testReportRun=1&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on release-24.3 @ [bbce415047c9896ee3b33b1eb4c06e3d2cab5bd6](https://github.com/cockroachdb/cockroach/commits/bbce415047c9896ee3b33b1eb4c06e3d2cab5bd6):


```
=== RUN   TestTenantZip
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantZip3333342508
    test_log_scope.go:76: use -show-logs to present logs inline
    zip_tenant_test.go:111: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantZip3333342508
--- FAIL: TestTenantZip (91.03s)
=== RUN   TestTenantZip/testzip_external_process_virtualization
[debug --host=127.0.0.1:42433 --insecure=true zip --concurrency=1 --cpu-profile-duration=1s /dev/null]
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/cli/cli_test_/cli_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/testdata/zip/testzip_external_process_virtualization:1:
         
        output didn't match expected:
        @@ -91,11 +91,13 @@
         [cluster] requesting tenant ranges: last request failed: rpc error: ...
         [cluster] requesting tenant ranges: creating error output: debug/tenant_ranges.err.txt... done
         [cluster] collecting the inflight traces for jobs... received response... done
         [cluster] requesting CPU profiles
         [cluster] profiles generated
        -[cluster] profile for node 1... writing binary output: debug/nodes/1/cpu.pprof... done
        +[cluster] profile for node 1...
        +[cluster] profile for node 1: last request failed: rpc error: ...
        +[cluster] profile for node 1: creating error output: debug/nodes/1/cpu.pprof.err.txt... done
         [node 1] node status... writing JSON output: debug/nodes/1/status.json... done
         [node 1] using SQL connection URL: postgresql://...
         [node 1] retrieving SQL data for crdb_internal.active_range_feeds... writing output: debug/nodes/1/crdb_internal.active_range_feeds.txt... done
         [node 1] retrieving SQL data for crdb_internal.cluster_replication_node_stream_checkpoints... writing output: debug/nodes/1/crdb_internal.cluster_replication_node_stream_checkpoints.txt... done
         [node 1] retrieving SQL data for crdb_internal.cluster_replication_node_stream_spans... writing output: debug/nodes/1/crdb_internal.cluster_replication_node_stream_spans.txt... done
    --- FAIL: TestTenantZip/testzip_external_process_virtualization (30.99s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-43963

--- a/pkg/cli/zip_tenant_test.go
+++ b/pkg/cli/zip_tenant_test.go
@@ -26,6 +26,7 @@ func TestTenantZip(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderRace(t, "test too slow under race")
+	skip.UnderDeadlock(t, "this test takes cpu profiles")
 
 	tenants := []struct {
 		testName      string

